morphia/src/test/java/org/mongodb/morphia/TestQuery.java

```diff
@@ -369,37 +391,67 @@ public void testElemMatchQueryCanBeNegated() {
+                                                        .elemMatch(getDs()
+                                                                       .createQuery(Keyword.class)
+                                                                       .field("keyword").equal("Scott"))
+                                                        .asKeyList());
+
+        validate2(pwk1, pwk2, pwk3, pwk4, mapper, getDs().find(PhotoWithKeywords.class)
+                                                         .field("keywords")
+                                                         .hasThisElement(new Keyword(14))
+                                                         .asKeyList());
+
+        validate2(pwk1, pwk2, pwk3, pwk4, mapper, getDs().find(PhotoWithKeywords.class)
+                                                         .field("keywords")
+                                                         .elemMatch(getDs()
+                                                                        .createQuery(Keyword.class)
+                                                                        .field("score").equal(14))
+                                                         .asKeyList());
+
+        validate3(pwk1, pwk2, pwk3, mapper, getDs().find(PhotoWithKeywords.class)
                                                    .field("keywords")
-                                                   .hasThisElement(new Keyword("Scott"))
-                                                   .asKeyList();
+                                                   .doesNotHaveThisElement(new Keyword("Scott"))
+                                                   .asKeyList());
 
-        assertEquals(2, keys.size());
-        assertFalse("because it doesn't have any keywords.", keys.contains(mapper.getKey(pwk1)));
-        assertFalse("because it doesn't have a matching keyword.", keys.contains(mapper.getKey(pwk2)));
-        assertTrue("because it has matching keyword.", keys.contains(mapper.getKey(pwk3)));
-        assertTrue("because it has matching keyword.", keys.contains(mapper.getKey(pwk4)));
+        validate3(pwk1, pwk2, pwk3, mapper, getDs().find(PhotoWithKeywords.class)
+                                                   .field("keywords").not()
+                                                   .elemMatch(getDs()
+                                                                  .createQuery(Keyword.class)
+                                                                  .field("keyword").equal("Scott"))
+                                                   .asKeyList());
+    }
 
-        keys = getDs().find(PhotoWithKeywords.class)
-                      .field("keywords")
-                      .hasThisElement(new Keyword(14))
-                      .asKeyList();
+    private void validate3(final PhotoWithKeywords pwk1, final PhotoWithKeywords pwk2, final PhotoWithKeywords pwk3, final Mapper mapper,
```

Comment:
validate1/2/3 are not descriptive enough names.   Try to describe the purpose of each one.